client/src/app/components/stardust/history/TransactionHistory.tsx

```diff
@@ -37,94 +44,50 @@ const TransactionHistory: React.FC<TransactionHistoryProps> = (
+                                transactionId={c.transactionId}
+                                isSpent={c.isSpent}
+                                isFormattedAmounts={isFormattedAmounts}
+                                setIsFormattedAmounts={setIsFormattedAmounts}
+                                darkBackgroundRow={idx % 2 === 0}
+                            />
+                        </React.Fragment>
+                    ))}
                 </tbody>
             </table>
 
-            {/* Only visible in mobile -- Card transactions*/}
+            {/** Only visible in mobile -- Card transactions */}
             <div className="transaction-history--cards">
-                {historyView.length > 0 && (
-                    historyView.map((historyItem, idx) => {
-                        const outputDetails = outputDetailsMap[historyItem.outputId];
-                        if (!outputDetails) {
-                            return null;
-                        }
-                        const transactionId = historyItem.isSpent ?
-                            outputDetails.metadata.transactionIdSpent :
-                            outputDetails.metadata.transactionId;
-
-                        if (!transactionId) {
-                            return null;
-                        }
-
+                    {transactions.map((c, idx) => {
                         return (
                             <React.Fragment key={idx}>
                                 <TransactionCard
-                                    outputId={historyItem.outputId}
-                                    transactionId={transactionId}
-                                    date={historyItem.milestoneTimestamp}
-                                    milestoneIndex={historyItem.milestoneIndex}
-                                    value={Number(outputDetails.output.amount)}
-                                    isSpent={historyItem.isSpent}
+                                    isGenesisByDate={c.isGenesisByDate}
+                                    isTransactionFromStardustGenesis={c.isTransactionFromStardustGenesis}
+                                    transactionLink={c.transactionLink}
+                                    dateFormatted={c.dateFormatted}
+                                    balanceChangeFormatted={c.balanceChangeFormatted}
+                                    transactionId={c.transactionId}
+                                    isSpent={c.isSpent}
                                     isFormattedAmounts={isFormattedAmounts}
                                     setIsFormattedAmounts={setIsFormattedAmounts}
                                 />
                             </React.Fragment>
                         );
-                    })
-                )}
+                    })}
             </div>
             {hasMore && historyView.length > 0 && (
                 <div className="card load-more--button" onClick={loadMore}>
```

Comment:
The load more logic should be improved too, it looks like load more related to outputs, not to number of txs, if its not too complicated, I would try to make it so it loads "N" more transactions, instead of "N" more outputs. @panteleymonchuk If this change gets too big, we can also tackle this in a different PR and get this to production before